### PR TITLE
Floodcheck contact form

### LINKED ISSUE #16: Contact Us Form - Floodchecking
**Is your feature request related to a problem? Please describe.**
Technically speaking, someone could currently put a bot to the contact-us form and spam our discord channel. So we should floodcheck it somehow

**Describe the solution you'd like**
Either have some form of captcha, but even better just time someone out for maybe 30-60 seconds after sending (could be stored in a cookie)

**Additional context**
If we're using cookies we probably should have a consent checkbox at the form that we'll be storing a cookie for X amount of time _🙄 @ GDPR_

### PR DESCRIPTION
**Pull request by justjustie**
resolve #16

### CODE REVIEW

**Comment by justjustie** [comment]
I cant view BetterCodeHub's complaints since its your repo :P

File: Hallowspeak/Pages/Contact.razor [hunk 1]
```diff
@@ -135,6 +149,10 @@
                 Message = string.Empty;
                 InvokeAsync(StateHasChanged);
             });
+
+            IpBlackList.Add(clientIp);
+
+            Task.Run(async () => { await Task.Delay(new TimeSpan(0, 1, 0)); IpBlackList.Remove(clientIp); });
```
Comment (dylanpiera) [inline Hallowspeak/Pages/Contact.razor]: Have you tested that if you close/leave the page this code still runs? I'm like 80% sure that it won't x)

File: Hallowspeak/Pages/Contact.razor [hunk 2]
```diff
@@ -78,6 +82,8 @@
     private (bool, string) ErrorMessage { get; set; }
     private (bool, string)? Success { get; set; }
 
+    private HashSet<IPAddress> IpBlackList = new HashSet<IPAddress>();
```
Comment (dylanpiera) [inline Hallowspeak/Pages/Contact.razor]: This is a local instance, if you refresh the page, it would be reset to null.

**Review by dylanpiera — CHANGES_REQUESTED** [review]
So couple of things to note, as I put in the specific comments in the code, your hashset is a local instance that when the page is refreshed is restored to blank. This should be a global singleton so you can't just refresh and retry.

I'm also pretty certain that if you refresh/close the page that the remove command won't be triggered. So either test that or think of a different solution for that one.

(I know a very simple one that I'd use but I'll let you tinker with it, if you can't think of anything just ping me)

File: Hallowspeak/Pages/Contact.razor [hunk 3]
```diff
@@ -2,8 +2,8 @@
 @page "/contact-us"
 @inject DiscordClientCredentials credentials
 @inject HttpContextAccessor httpContextAccessor
-@using System.Collections.Generic
-@using System.Net
+@inject Floodchecker floodchecker
+@using System.Net;
```
Comment (justjustie) [inline Hallowspeak/Pages/Contact.razor]: Rogue semicolon here :p

File: Hallowspeak/Pages/Contact.razor [hunk 4]
```diff
@@ -1,6 +1,10 @@
 ﻿@page "/contact"
 @page "/contact-us"
 @inject DiscordClientCredentials credentials
+@inject HttpContextAccessor httpContextAccessor
+@inject Floodchecker floodchecker
+@using System.Net;
```
Comment (dylanpiera) [inline Hallowspeak/Pages/Contact.razor]:
```suggestion
@using System.Net
```

**Review by dylanpiera — CHANGES_REQUESTED** [review]
Solves one issue, but did you test the other?

**Comment by justjustie** [comment]
This might be a little over the top but it works as far as I can tell and I'm kinda proud of it haha

**Comment by dylanpiera** [comment]
I don't think it's over the top, but you've tested and made sure that even when refreshed it still blocks, and after a minute from the original request it times out?

**Comment by justjustie** [comment]
I tested, still blocked after refresh, unblocks after a refresh. Can't really think of more testing...

**Pull request merged**